File: crates/dewasm-backend-go/units/table/copy.go
// requires: rt/trap, table/check_range, table/slice
func (t *Table) copy(dst uint32, other *Table, src, length uint32) {
    if uint64(src)+uint64(length) > uint64(other.size()) {
        Rt.trap("out of bounds table access")
    }
    t.check_range(dst, length)
    if length == 0 {
        return
    }
    copy(t.slots[dst:dst+length], other.slice(src, length))
}

File: crates/dewasm-backend-ruby/units/table/copy.rb
# requires: rt/trap, table/check_range, table/slice
def copy(dst, other, src, len)
  Rt.trap("out of bounds table access") if src + len > other.size
  check_range(dst, len)
  return if len == 0
  @slots[dst, len] = other.slice(src, len)
end
